res/controllers/PIONEER-DDJ-ERGO-scripts.js

```diff
@@ -0,0 +1,441 @@
+    if (newValue > 0) {
+        engine.setValue(group, "beats_translate_later", 1);
+    }
+    if (newValue < 0) {
+        engine.setValue(group, "beats_translate_earlier", 1);
+    }
+};
+
+//this opens/closes directories when browsing the sidebar and shift + turn the browse wheel
+DDJERGO.browseSidebar = function(channel, control, value) {
+    let newValue;
+    if (value-64 > 0) { newValue = value-128; } else { newValue = value; }
+    if (newValue > 0) {
+        engine.setParameter("[Library]", "MoveRight", 1);
+    }
+    if (newValue < 0) {
+        engine.setParameter("[Library]", "MoveLeft", 1);
+    }
+};
+
+//this stores the FX button presses in the selectedEffects variable and enables/disables the effect with the on fx button click
+//this also deals with LEDs, turning on the FX slot button currently active and the ON button
+DDJERGO.effectsFunction = function(channel, control, value, status) {
+    const deck = status-0x94;
+    const effect = selectedEffects[deck];
+    //on button pressed
+    if (control === 0x43 && value === 0x7F) {
+        const toggleState = 1 - engine.getParameter(effectsStrings[deck][effect], "enabled");
+        engine.setParameter(effectsStrings[deck][effect], "enabled", toggleState);
+        //toggle ON button and selected effect button
+        midi.sendShortMsg(status, 0x43, 0x7F*toggleState);
+        midi.sendShortMsg(status, 0x47+effect, 0x7F*toggleState);
+    }
+    //fx number button pressed
+    else if (control >= 0x47 && value === 0x7F) {
+        selectedEffects[deck] = control - 0x47;
+    }
+};
+
+//this modifies the selected effects parameters based on the MSBs of the FX knobs
+DDJERGO.effectsParams = function(channel, control, value, status) {
+    const deck = status-0xB4;
+    const effect = selectedEffects[deck];
+    //if called with MSB
+    if (control < 0x20) {
+        //if fx selector turned enough
+        if (control === 0x00 && previousEffectKnobPosition !== value/5) {
+            //switch to next or previous fx in list
+            if (previousEffectKnobPosition > value/5) {
+                engine.setValue(effectsStrings[deck][effect], "effect_selector", -1);
+            } else {
+                engine.setValue(effectsStrings[deck][effect], "effect_selector", 1);
+            }
+            previousEffectKnobPosition = value/5;
+        }
+        //if fx knob 1 turned
+        if (control === 0x02) {
+            engine.setValue(effectsStrings[deck][effect], "meta", value/127);
+        }
+    }
+};
+
+//this plays the samples on the samplers with a click on the samplers buttons and stores its value for the sample vol knob to manipulate it
+DDJERGO.samplers = function(channel, control, value, status, group) {
+    const deck = status-0x90;
+    if (value === 0x7F) {
+        if (engine.getParameter(group, "play") === 1) {
+            engine.setParameter(group, "cue_default", 1);
+        } else {
+            engine.setParameter(group, "play", 1);
+        }
```

Comment:
```suggestion
        if (engine.getParameter(group, "play") === 1) {
            engine.setParameter(group, "cue_gotoandstop", 1);
        } else {
            engine.setParameter(group, "cue_gotoandplay", 1);
        }
```
`cue_gotoandplay` only to make sure you play from cue in case you dropped a track from another deck (clone) and the play pos is not at the cue.